cmd/soroban-rpc/internal/methods/transaction.go

```diff
@@ -277,8 +280,11 @@ func (p *TransactionProxy) GetTransactionStatus(ctx context.Context, request Get
 		if herr, ok := err.(*horizonclient.Error); ok {
 			if herr.Problem.Status != http.StatusNotFound {
 				return TransactionStatusResponse{
-					ID:     request.Hash,
-					Status: TransactionError,
+					ID:            request.Hash,
```

Comment:
just observation, maybe it's time for `TransactionStatusResponse.New(id string, status string, tx horizon.Transaction, err *TransactionResponseError )`